ibm/resource_ibm_enterprise_account_test.go

```diff
@@ -0,0 +1,149 @@
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/resource"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/terraform"
+	"testing"
+
+	"github.com/IBM/platform-services-go-sdk/enterprisemanagementv1"
+)
+
+func TestAccIbmEnterpriseAccountBasic(t *testing.T) {
+	var conf enterprisemanagementv1.Account
+	//parent := fmt.Sprintf("parent_%d", acctest.RandIntRange(10, 100))
+	name := fmt.Sprintf("tf-gen-account-name_%d", acctest.RandIntRange(10, 100))
+	//ownerIamID := fmt.Sprintf("owner_iam_id_%d", acctest.RandIntRange(10, 100))
+	//parentUpdate := fmt.Sprintf("parent_%d", acctest.RandIntRange(10, 100))
+
+	resource.Test(t, resource.TestCase{
+		PreCheck:  func() { testAccPreCheckEnterprise(t) },
+		Providers: testAccProviders,
+		Steps: []resource.TestStep{
+			resource.TestStep{
+				Config: testAccCheckIbmEnterpriseAccountConfigBasic(name),
+				Check: resource.ComposeAggregateTestCheckFunc(
+					testAccCheckIbmEnterpriseAccountExists("ibm_enterprise_account.enterprise_account", conf),
+					resource.TestCheckResourceAttrSet("ibm_enterprise_account.enterprise_account", "parent"),
+					resource.TestCheckResourceAttr("ibm_enterprise_account.enterprise_account", "name", name),
+					resource.TestCheckResourceAttrSet("ibm_enterprise_account.enterprise_account", "owner_iam_id"),
+				),
+			},
+			resource.TestStep{
+				Config: testAccCheckIbmEnterpriseAccountConfigUpdateBasic(name),
+				Check: resource.ComposeAggregateTestCheckFunc(
+					resource.TestCheckResourceAttrSet("ibm_enterprise_account.enterprise_account", "parent"),
+					resource.TestCheckResourceAttrSet("ibm_enterprise_account.enterprise_account", "name"),
+					resource.TestCheckResourceAttrSet("ibm_enterprise_account.enterprise_account", "owner_iam_id"),
+				),
+			},
+			resource.TestStep{
+				ResourceName:      "ibm_enterprise_account.enterprise_account",
+				ImportState:       true,
+				ImportStateVerify: true,
+			},
+		},
+	})
+}
+
+func TestAccIbmEnterpriseImportAccountBasic(t *testing.T) {
+	var conf enterprisemanagementv1.Account
+	resource.Test(t, resource.TestCase{
+		PreCheck:  func() { testAccPreCheckEnterpriseAccountImport(t) },
+		Providers: testAccProviders,
+		Steps: []resource.TestStep{
+			resource.TestStep{
+				Config: testAccCheckIbmAccountsDataSourceConfigImportBasic(account_to_be_imported),
+				Check: resource.ComposeAggregateTestCheckFunc(
+					testAccCheckIbmEnterpriseAccountExists("ibm_enterprise_account.enterprise_account_import", conf),
+					resource.TestCheckResourceAttrSet("ibm_enterprise_account.enterprise_account_import", "parent"),
+					resource.TestCheckResourceAttr("ibm_enterprise_account.enterprise_account_import", "account_id", account_to_be_imported),
+					resource.TestCheckResourceAttrSet("ibm_enterprise_account.enterprise_account_import", "owner_iam_id"),
+				),
+			},
+		},
+	})
+}
+
+func testAccCheckIbmEnterpriseAccountConfigBasic(name string) string {
+	return fmt.Sprintf(`
+		data "ibm_enterprises" "enterprises_instance" {
```

Comment:
you are using datasource to read enterprise so here do we have any prereq we should have enterprises already ccreated

Reply:
Yes, for account and account_group test we should be using the IC_API_KEY of an already existing enterprise account.